casaos-healthcheck/run.sh

```
@@ -381,6 +381,325 @@ check_dmesg_errors() {
+check_time_sync() {
+    print_header "Time Synchronization Check"
+    
+    if command -v timedatectl &>/dev/null; then
+        local ntp_status=$(timedatectl | grep "NTP synchronized")
+        if [[ $ntp_status == *"yes"* ]]; then
+            print_color "0;32" "${CHECK_MARK} NTP is synchronized"
+        else
+            print_color "0;31" "${CROSS_MARK} NTP is not synchronized"
+        fi
+        
+        local time_status=$(timedatectl status --no-pager)
```

Shellcheck

[warning] Declare and assign separately to avoid masking return values.

(SC2155)